### PR TITLE
[build] fixes for ot-efr32 builds

### PR DESCRIPTION
**Pull request by suveshpratapa**
- OPENTHREAD_CONFIG_NCP_UART_ENABLE is now OPENTHREAD_CONFIG_NCP_HDLC_ENABLE
- Pull in cli_uart.cpp UART API interface for sleepy-demo apps.
- Fix some erroneous ifdefs that affect 1.2 SoC builds and 1.1 builds.
- Add OT_RADIO_CAPS_SLEEP_TO_TX to radio capabilities.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -57,7 +57,7 @@ target_compile_definitions(ot-config INTERFACE
     OPENTHREAD_CONFIG_FILE="openthread-core-efr32-config.h"
     OPENTHREAD_PROJECT_CORE_CONFIG_FILE="openthread-core-efr32-config.h"
     OPENTHREAD_CORE_CONFIG_PLATFORM_CHECK_FILE="openthread-core-efr32-config-check.h"
-    OPENTHREAD_CONFIG_NCP_UART_ENABLE=1
+    OPENTHREAD_CONFIG_NCP_HDLC_ENABLE=1
```
Comment (lmnotran) [inline CMakeLists.txt:60]: Is this line redundant with the define in [src/src/openthread-core-efr32-config.h](https://github.com/openthread/ot-efr32/blob/23c69b7302682666b93fa253128a9c3ffd48472d/src/src/openthread-core-efr32-config.h)?

Comment (suveshpratapa) [reply CMakeLists.txt:60]: Could be, I just made the rename changes in place wherever I found it.  Let me test it with it removed.

Comment (suveshpratapa) [reply CMakeLists.txt]: Yes, it was redundant. Removed the line.  Thanks for catching that!

**Review by lmnotran — APPROVED** [review]

**Comment by suveshpratapa** [comment]
This PR should address https://github.com/openthread/openthread/issues/5819

**Review by romacdon — APPROVED** [review]
LGTM

**Review by jwhui — APPROVED** [review]

**Comment by jwhui** [comment]
@suveshpratapa , can you help resolve the submodule conflict?

File: src/src/uart.c [hunk 2]
```diff
@@ -36,7 +36,7 @@
 
 #include "openthread-system.h"
 #include <openthread-core-config.h>
-#include <openthread/platform/uart.h>
+#include "utils/uart.h"
 
 #include "utils/code_utils.h"
```
Comment (jwhui) [inline src/src/uart.c:41]: nit:
```suggestion

#include "utils/code_utils.h"
#include "utils/uart.h"
```

File: src/src/system.c [hunk 3]
```diff
@@ -39,7 +39,7 @@
 #include <string.h>
 
 #include "openthread-system.h"
-#include <openthread/platform/uart.h>
+#include "utils/uart.h"
 
 #include "common/logging.hpp"
```
Comment (jwhui) [inline src/src/system.c:44]: nit:
```suggestion

#include "common/logging.hpp"
#include "utils/uart.h"
```

Comment (suveshpratapa) [reply src/src/system.c]: Thanks, fixed.

File: src/src/uart.c (hunk 2 above)
Comment (suveshpratapa) [reply src/src/uart.c]: Thanks, fixed.

**Comment by suveshpratapa** [comment]
> @suveshpratapa , can you help resolve the submodule conflict?

Should be resolved now, let me know if there are any issues.

**Comment by jwhui** [comment]
> Should be resolved now, let me know if there are any issues.

@suveshpratapa , looks good, thanks!

**Pull request merged**